authentication/src/main/java/io/camunda/authentication/entity/CamundaOidcUser.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.entity;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.OidcUserInfo;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+
+public class CamundaOidcUser implements OidcUser, CamundaPrincipal {
+  private final OidcUser user;
+  private final AuthenticationContext authentication;
+  private final Set<Long> mappingKeys;
+
+  public CamundaOidcUser(
+      final OidcUser oidcUser,
+      final Set<Long> mappingKeys,
+      final AuthenticationContext authentication) {
+    user = oidcUser;
+    this.mappingKeys = mappingKeys;
+    this.authentication = authentication;
+  }
+
+  @Override
+  public String getEmail() {
+    return user.getEmail();
+  }
+
+  @Override
+  public String getDisplayName() {
+    return user.getClaim("nickname");
```

Comment:
❓ @emulbreh - is this always `nickname`? Should it be configurable?

Reply:
I don't know what this should be, but hardcoding the claim name definitely isn't right. There's probably some overlap with the "simple mapping" configuration which also needs configuration for well-known claims.

Reply:
For now I've reverted it to `user.getPreferredUsername()`